src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java

```diff
@@ -425,6 +430,19 @@ protected void checkImageWithAutoPull(ServiceHub hub, String image, String regis
         }
     }
 
+    private synchronized Set<String> getPreviouslyPulledImageCache() {
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        Set<String> cache = (Set) session.getUserProperties().get(CONTEXT_KEY_PREVIOUSLY_PULLED);
+        if (cache == null) {
+            System.out.println("creating new context");
+
+            cache = Sets.newConcurrentHashSet();
```

Comment:
Any reason why a simple HashSet() doesn't work (or Hashtable() if it needs to be synchronized) ? Just curious whether the google collections provides some additional benefits. not a principal problems, since we already have this dependency ....